### PR TITLE
fix(stream): fix `XINFO` group metadata key overlapping.

### LINKED ISSUE #2568: XINFO GROUPS fails to deduplicate; returns invalid `pending`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

unstable commit b8cf118bbe56984c7112c0929402f79d81d8a0d7

### Minimal reproduce step

```
127.0.0.1:3847> FLUSHALL
OK
127.0.0.1:3847> XGROUP CREATE stream1 group 0 MKSTREAM
OK
127.0.0.1:3847> XGROUP CREATE stream2 group 0 MKSTREAM
OK
127.0.0.1:3847> XREADGROUP GROUP group consumer STREAMS stream1 stream2 0 0
(nil)
127.0.0.1:3847> XINFO GROUPS stream2
1)  1) "name"
    2) "group"
    3) "consumers"
    4) (integer) 1
    5) "pending"
    6) (integer) 0
    7) "last-delivered-id"
    8) "0-0"
    9) "entries-read"
   10) (nil)
   11) "lag"
   12) (integer) 0
2)  1) "name"
    2) "group"
    3) "consumers"
    4) (integer) 0
    5) "pending"
    6) (integer) 1727818077688
    7) "last-delivered-id"
    8) "1727818077688-0"
    9) "entries-read"
   10) (integer) 0
   11) "lag"
   12) (integer) 0
```

### What did you expect to see?

The following output comes from Redis:
```
127.0.0.1:6379> FLUSHALL
OK
127.0.0.1:6379> XGROUP CREATE stream1 group 0 MKSTREAM
OK
127.0.0.1:6379> XGROUP CREATE stream2 group 0 MKSTREAM
OK
127.0.0.1:6379> XREADGROUP GROUP group consumer STREAMS stream1 stream2 0 0
1) 1) "stream1"
   2) (empty array)
2) 1) "stream2"
   2) (empty array)
127.0.0.1:6379> XINFO GROUPS stream2
1) 1) "name"
   2) "group"
   3) "consumers"
   4) (integer) 1
   5) "pending"
   6) (integer) 0
   7) "last-delivered-id"
   8) "0-0"
```

There are a few notable differences here:
1. `XREADGROUP GROUP` returns `stream1` and `stream2` as empty arrays instead of returning `nil` for the entire request
2. `XINFO GROUPS stream2` doesn't return a duplicated consumer group
3. `XINFO GROUPS stream2` correctly returns `pending = 0`

### What did you see instead?

`XINFO GROUPS` returns duplicated consumer groups, and returns an incorrect value for `pending`. 

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by LindaSummer**
# Issue

Fix #2568 

# Root cause

Here is the group metadata subkey.
https://github.com/apache/kvrocks/blob/0a43bade1a4c3c885947fdd42eacefac934bf1bd/src/types/redis_stream.cc#L172-L181

Here is the consumer metadata subkey.
https://github.com/apache/kvrocks/blob/0a43bade1a4c3c885947fdd42eacefac934bf1bd/src/types/redis_stream.cc#L222-L233

The enum `StreamConsumerGroupMetadata` is 1 and `StreamConsumerMetadata` is 2.

Here is group info key scanning.
https://github.com/apache/kvrocks/blob/0a43bade1a4c3c885947fdd42eacefac934bf1bd/src/types/redis_stream.cc#L1343-L1348

When search the whole version from `StreamConsumerGroupMetadata`, the `StreamConsumerMetadata` will be included.

Here is an example below.

```
0b5f5f6e616d6573706163650000000773747265616d32311cf46497d771f3ffffffffffffffff01 # lower bound

0b5f5f6e616d6573706163650000000773747265616d32311cf46497d771f3ffffffffffffffff01000000000000000567726f7570 # consumer group metadata
0b5f5f6e616d6573706163650000000773747265616d32311cf46497d771f3ffffffffffffffff02000000000000000567726f75700000000000000008636f6e73756d6572 # consumer metadata


0b5f5f6e616d6573706163650000000773747265616d32311cf46497d771f4ffffffffffffffff01 # upper bound
```

So, we should filter the key we got when searching through the whole version.

By the way, there is another bug.

In encoding, we use `UINT64_MAX` as bytes rather than literal string for `subkey_type_delimiter`.
```
encode: PutFixed64(&sub_key, UINT64_MAX);

search: std::string subkey_type_delimiter = std::to_string(UINT64_MAX);
```

It should also be fixed if I understand correctly.

# Proposed Changes

- Use `identifySubkeyType` to filter metadata key for searching.
- Fix `subkey_type_delimiter` encoding bug.
- Add this issue to unit test.

### CODE REVIEW

**Comment by nathanlo-hrt** [comment]
thanks for taking a look at this!

File: src/types/redis_stream.cc [hunk 1]
```diff
@@ -1340,7 +1340,8 @@ rocksdb::Status Stream::GetGroupInfo(engine::Context &ctx, const Slice &stream_n
   rocksdb::Status s = GetMetadata(ctx, ns_key, &metadata);
   if (!s.ok()) return s;
 
-  std::string subkey_type_delimiter = std::to_string(UINT64_MAX);
+  std::string subkey_type_delimiter;
+  PutFixed64(&subkey_type_delimiter, UINT64_MAX);
```
Comment (PragmaTwice) [inline src/types/redis_stream.cc:1344]: Good catch!

@git-hulk That's seems a very serious bug in stream group which leads to wrong data encoding.
Should we consider to cancel the current release vote (2.10.0) for it?

**Review by PragmaTwice — APPROVED** [review]

**Review by mapleFU — APPROVED** [review]

**Comment by jonathanc-n** [comment]
@LindaSummer Nice change!

**Comment by Yangsx-1** [comment]
@LindaSummer Good catch!!!

**Pull request merged**